scripts/utils.py

```diff
@@ -16,28 +17,34 @@ def extract_tar_file_contents(filename, filetype):
     """
     extension = EXTENSION_BY_FILETYPE[filetype]
 
-    tar = tarfile.open(filename)
-    internal_file = None
-    for member in tar.getmembers():
-        suffixes = Path(member.name).suffixes
+    with tarfile.open(filename) as tar:
+        internal_file = None
+        for member in tar.getmembers():
+            suffixes = Path(member.name).suffixes
 
-        if extension in suffixes:
-            # By default, return the binary stream for the member file.
-            internal_file = tar.extractfile(member.name)
+            if extension in suffixes:
+                # By default, return the binary stream for the member file.
+                internal_file = tar.extractfile(member.name)
 
-            if ".xz" in suffixes:
-                # Check for LZMA-compressed data and open these with the
-                # corresponding library.
-                internal_file = lzma.open(internal_file, "rt")
-            elif extension == ".fasta":
-                # For sequence data, handle decoding of the binary stream prior
-                # to passing the data back to the caller.
-                internal_file = (line.decode("utf-8") for line in internal_file)
+                if ".xz" in suffixes:
+                    # Check for LZMA-compressed data and open these with the
+                    # corresponding library.
+                    internal_file = lzma.open(internal_file, "rt")
```

Comment:
I realize the mode here is historic, but why do `.xz` files default to decoding as text? Esp. since later any text streams are re-encoded (presumably back) to UTF-8?

Reply:
Good eye on this one and the comment below, @tsibley! The text stream was because the calling code used to consume the returned object from this function directly with BioPython and BioPython expected a text stream. There is no reason to do this extra work now.

As you pointed out below, there's also no reason to write out the contents of these files line by line! I rewrote this logic in 79497a1 to use `extractall` to extract the files into a temporary directory (with their original names) that the calling code can clean up (and that Python will automatically clean up when the temp dir objects get destroyed). These changes make the tar extraction way simpler and also keeps the files in compressed format on disk, etc. I think this is much nicer, but I'm curious what you think about this revision.

Reply:
Awesome. I like this simpler way of doing it.

Since `extract_tar_file_contents()` no longer transparently decompresses `.xz` files, I assume all callers/downstream uses can handle compressed files? IIRC, it was only BioPython which couldn't directly accept an `.xz` file?

Reply:
Yeah, support for compressed files was added a while back in augur and the ncov workflow. The reason not to use compressed files previously was that we were streaming the contents from the tar file.